arbnode/espresso_caff_node.go

```diff
@@ -354,7 +354,7 @@ func (n *EspressoCaffNode) nextMessage() (*espressostreamer.MessageWithMetadataA
 	}
 
 	if messageWithMetadataAndPos == nil {
-		return nil, nil
+		return nil, fmt.Errorf("espressoStreamer had no message to produce")
```

Comment:
if the result is nil, it means the streamer doesn't have next message yet. So I don't think this is an error, it is just an empty value.

In most of the api design, next function should retrun an Option<T> but not Result<T, Error>

Reply:
I agree that the Option semantics are better!